### PR TITLE
Add FileUtils#cp_r and #rm_r

### PR DESCRIPTION
**Pull request by Dreauw**
Adds cp_r and rm_r to copy or remove a directory (like in Ruby)

No FileUtils#rm as it would be an alias of File#delete

### CODE REVIEW

File: spec/std/file_utils_spec.cr [hunk 1]
```diff
@@ -51,4 +51,54 @@ describe "FileUtils" do
...
+        File.exists?(dest_path + "b/c").should be_true
+      ensure
+        File.delete(dest_path + "b/c") if File.exists?(dest_path + "b/c")
+        File.delete(dest_path + "a") if File.exists?(dest_path + "a")
+        Dir.rmdir(dest_path + "b") if Dir.exists?(dest_path)
+        Dir.rmdir(dest_path) if Dir.exists?(dest_path)
+        File.delete(src_path + "b/c") if File.exists?(src_path + "b/c")
+        File.delete(src_path + "a") if File.exists?(src_path + "a")
+        Dir.rmdir(src_path + "b") if Dir.exists?(src_path)
+        Dir.rmdir(src_path) if Dir.exists?(src_path)
```
Comment (ysbaddaden) [inline spec/std/file_utils_spec.cr]: What about using `FileUtils.rm_r` for cleanup?

**Comment by Dreauw** [comment]
@ysbaddaden 
Yeah, using FileUtils.rm_r for cleanup would be a lot easier, but the method hasn't been merged yet (so you could decide to accept cp_r but not rm_r) and I don't like the idea of relying on such a "big" method in the tests.
I feel more comfortable using simple methods like File#delete and Dir#rmdir (as they are unlikely to fail).

But I can change it if you really want it.

**Comment by jhass** [comment]
LGTM, @ysbaddaden did you revise your stance?

**Comment by ysbaddaden** [comment]
Let's squash commits as a single one and use `FileUtils.rm_r` in the ensure cleanup to simplify the spec?

**Comment by jhass** [comment]
I don't mind either really, will do.

**Comment by jhass** [comment]
Merged as 55ce2a86580b4ea540e7b053230b6013456f5edf, thanks!

**Pull request closed without merging**